cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemotePluginFinder.java

```diff
@@ -144,6 +144,8 @@ public RemotePluginFinder(LocationFactory locationFactory, RemoteClientFactory r
                                       String pluginType,
                                       String pluginName)
     throws IOException, PluginNotExistsException, UnauthorizedException {
+    // replace the space in the name 
+    pluginName = pluginName.replace(" ", "%20");
```

Comment:
manually replace it looks a little bit weird, there could be other special character that needs to be encoded. why not encoding the whole `namespaces/%s/artifacts/%s/versions/%s/extensions/%s/plugins/%s` ?

Reply:
namespace, artifact, version and plugin type I believe we have some restrictions on it so special characters cannot be there. Encode the whole url will make the call on data fusion fail, same reason as the problem we fix in #2.